pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the ternary discrete-variable quantum operations
+that do not depend on any parameters.
+"""
+# pylint:disable=arguments-differ
+import numpy as np
+
+import pennylane as qml  # pylint: disable=unused-import
+from pennylane.operation import Operation
+
+OMEGA = np.exp(2 * np.pi * 1j / 3)
+ZETA = OMEGA ** (1 / 3)  # ZETA will be used as a phase for later non-parametric operations
+
+# Note: When Operation.matrix() is used for qutrit operations, `wire_order` must be `None` as specifying
+# an order that is expected to return a permuted and expanded matrix different from the canonical matrix
+# will lead to errors as `expand_matrix()` in `pennylane/operation.py`, which is used to compute the
+# permuted and expanded matrix from the canonical matrix, is hard coded to work correctly for qubits.
+
+
+class TShift(Operation):
+    r"""TShift(wires)
+    The qutrit shift operator
+
+    .. math:: TShift = \begin{bmatrix}
+                        0 & 0 & 1 \\
+                        1 & 0 & 0 \\
+                        0 & 1 & 0
+                    \end{bmatrix}
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_wires = 1
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
```

Comment:
It looks like operations also include a `label` method now; I assume this is just what's printed out in the circuit diagrams? I wonder what would be a good choice here.

Edit: Looks like there is also `_queue_category`; I am not sure though what that is.

Reply:
The `label` method is only overriden when a custom label is needed. For example, the base `label` method adds a inverse symbol after the name of the operation if the operation is adjointed. In this case, we don't need to override the base method as it already works out of the box. It would, however, need to be overridden for operations that are self-adjoint because the inverse symbol would be redundant.

As far as `_queue_category` goes, I'm not entirely sure what it is either.

Reply:
> It looks like operations also include a label method now; I assume this is just what's printed out in the circuit diagrams?

Yep, @albi3ro likely has the best context here :)

Reply:
`label` controls how the operation looks in text and matplotlib circuit diagrams.  `Operator` and `Operation` have good defaults, so if you are happy with how it looks, you can just leave it be.

`_queue_category` controls how the operator is processed into a tape. 
* `"_prep"`  indicates state preparation, and these must be at the beginning of a circuit
* `"_ops"` is any general operation in the middle of a circuit
* `"_measurements"` indicates that the object is a measurement. These have to be at the end.
* `None` indicates that the object should not be processed into any tape list. Objects like this tend to be observables like `Tensor`, `Hamiltonian`, `Projector`, etc.

Reply:
Awesome, thanks @albi3ro for the clear explanation!